src/ngraph/graph_util.cpp

```diff
@@ -153,14 +153,20 @@ void ngraph::replace_node(std::shared_ptr<Node> target,
     {
         auto common_args = ngraph::find_common_args(target, replacement);
 
-        auto set_replacement_prov = [replacement](std::shared_ptr<Node> node) {
-            replacement->merge_provenance_tags_from(node);
+        std::unordered_set<string> removed_subgraph_tags;
```

Comment:
Use an ordered set so that results are consistent on different runs.

Reply:
@diyessi Provenance tags (m_provenance_tags) is unordered. I am not sure making this one sorted would be enough.